Simplified/Adobe DRM/DataByteStream.cpp

```diff
@@ -0,0 +1,106 @@
+//
+//  DataByteStream.cpp
+//  SimplyE
+//
+//  Created by Vladimir Fedorov on 13.05.2020.
+//  Copyright © 2020 NYPL Labs. All rights reserved.
+//
+
+#include <stdio.h>
+#include "DataByteStream.h"
+
+DataByteStream::DataByteStream(unsigned char *bytes, unsigned long len) {
+    data = (unsigned char *)malloc(len);
+    memcpy(data, bytes, len);
+    length = len;
+    position = 0;
+}
+DataByteStream::~DataByteStream() {
+    free(data);
+}
+
+DataByteStream::size_type DataByteStream::BytesAvailable() _NOEXCEPT {
+    return length - position;
+}
+DataByteStream::size_type DataByteStream::SpaceAvailable() const _NOEXCEPT {
```

Comment:
what's semantically the difference between `BytesAvailable` and `SpaceAvailable`? Probably should go in method documentation. Also, why are always returning 0?

Reply:
BytesAvailable returns the number of bytes left in the stream for reading; SpaceAvailable returns number of bytes left for writing. We don't write to the stream (and can not, as this is an array of bytes for decryption), it always returns 0